Parse/src/main/java/com/parse/ParseOkHttpClient.java

```diff
@@ -146,6 +156,124 @@ public ParseOkHttpClient(int socketOperationTimeout, SSLSessionCache sslSessionC
+    // Set method
+    switch (okHttpRequest.method()) {
+       case OKHTTP_GET:
+           parseRequestBuilder.setMethod(ParseRequest.Method.GET);
+           break;
+       case OKHTTP_DELETE:
+           parseRequestBuilder.setMethod(ParseRequest.Method.DELETE);
+           break;
+       case OKHTTP_POST:
+           parseRequestBuilder.setMethod(ParseRequest.Method.POST);
+           break;
+       case OKHTTP_PUT:
+           parseRequestBuilder.setMethod(ParseRequest.Method.PUT);
+           break;
+       default:
+           // This should never happen
+           throw new IllegalArgumentException(
+               "Invalid http method " + okHttpRequest.method());
+     }
+
+    // Set url
+    parseRequestBuilder.setUrl(okHttpRequest.urlString());
+
+    // Set Header
+    for (Map.Entry<String, List<String>> entry : okHttpRequest.headers().toMultimap().entrySet()) {
+      parseRequestBuilder.addHeader(entry.getKey(), entry.getValue().get(0));
+    }
+
+    // Set Body
+    if (okHttpRequest.body() instanceof CountingOkHttpRequestBody) {
+      CountingOkHttpRequestBody okHttpBody = (CountingOkHttpRequestBody) okHttpRequest.body();
+      parseRequestBuilder.setBody(okHttpBody.getParseHttpBody());
+    }
+    return parseRequestBuilder.build();
+  }
+
+  /**
+   * For OKHttpClient, since it does not expose any interface for us to check the raw response
+   * stream, we have to use OKHttp networkInterceptors. Instead of using our own interceptor list,
+   * we use OKHttp inner interceptor list.
+   * @param parseNetworkInterceptor
+   */
+  @Override
+  /* package */ void addExternalInterceptor(final ParseNetworkInterceptor parseNetworkInterceptor) {
+    okHttpClient.networkInterceptors().add(new Interceptor() {
+      @Override
+      public Response intercept(final Chain okHttpChain) throws IOException {
+        Request okHttpRequest = okHttpChain.request();
+        // Transfer OkHttpRequest to ParseHttpRequest
+        final ParseHttpRequest parseRequest = getParseHttpRequest(okHttpRequest);
+        // Capture OkHttpResponse
+        final Capture<Response> okHttpResponseCapture = new Capture<>();
+        final ParseHttpResponse parseResponse =
+            parseNetworkInterceptor.intercept(new ParseNetworkInterceptor.Chain() {
+          @Override
+          public ParseHttpRequest getRequest() {
+            return parseRequest;
+          }
+
+          @Override
+          public ParseHttpResponse proceed(ParseHttpRequest parseRequest) throws IOException {
+            // Use OKHttpClient to send request
+            Request okHttpRequest = ParseOkHttpClient.this.getRequest(parseRequest);
+            Response okHttpResponse = okHttpChain.proceed(okHttpRequest);
+            okHttpResponseCapture.set(okHttpResponse);
+            return getResponse(okHttpResponse);
+          }
+        });
+        final Response okHttpResponse = okHttpResponseCapture.get();
```

Comment:
Yep, we can guarantee this, since `chain.proceed()` will always be called in `parseNetworkInterceptor.intercept()`.